libs/sdk-ui-kit/src/List/InvertableSelect/InvertableSelectVirtualised.tsx

```diff
@@ -0,0 +1,416 @@
+            getIsItemSelected,
+            getItemTitle,
+            selectItems,
+            deselectItems,
+            selectOnly,
+            focusedAction,
+            focusedItem,
+        ],
+    );
+
+    return (
+        <div className="gd-invertable-select">
+            <div className="gd-invertable-select-search-bar" onKeyDown={(e) => e.stopPropagation()}>
+                {renderSearchBar({ onSearch, searchPlaceholder, searchString })}
+            </div>
+            {isLoading ? (
+                <div className="gd-invertable-select-loading">{renderLoading({ height })}</div>
+            ) : error ? (
+                <div className="gd-invertable-select-error">{renderError({ height, error })}</div>
+            ) : (
+                <>
+                    {renderActions({
+                        isVisible: items.length > 0,
+                        checked: selectionState !== "none",
+                        onToggle: onSelectAllCheckboxToggle,
+                        onChange: onSelectAllCheckboxChange,
+                        isFiltered: searchString?.length > 0,
+                        totalItemsCount,
+                        isPartialSelection: selectionState === "partial",
+                    })}
+                    {items.length > 0 && (
+                        <Measure client>
+                            {({ measureRef, contentRect }) => {
+                                return isLoading ? (
+                                    <LoadingMask height={height} />
+                                ) : (
+                                    <div className="gd-invertable-select-list" ref={measureRef}>
+                                        <div
+                                            tabIndex={0}
+                                            onKeyDown={onKeyboardNavigation}
+                                            className={cx("gd-async-list", className ? className : "")}
+                                        >
+                                            <UiPagedVirtualList
+                                                items={items}
+                                                itemHeight={itemHeight}
+                                                itemsGap={0}
+                                                itemPadding={0}
+                                                skeletonItemsCount={nextPageItemPlaceholdersCount}
+                                                hasNextPage={nextPageItemPlaceholdersCount > 0}
+                                                loadNextPage={onLoadNextPage}
+                                                isLoading={isLoadingNextPage}
+                                                maxHeight={
+                                                    adaptiveHeight
+                                                        ? contentRect?.client.height
+                                                        : Math.min(items.length, 10) * itemHeight
```

Comment:
Can you extract the magic number as a constant?

Reply:
Done.